cpp/src/arrow/util/value_parsing.cc

```diff
@@ -43,46 +44,43 @@ struct StringToFloatConverter::Impl {
   util::double_conversion::StringToDoubleConverter fallback_converter_;
 };
 
-constexpr int StringToFloatConverter::Impl::flags_;
-constexpr double StringToFloatConverter::Impl::main_junk_value_;
-constexpr double StringToFloatConverter::Impl::fallback_junk_value_;
+static StringToFloatConverterImpl g_string_to_float;
```

Comment:
To make immutability/thread safety clearer:
```suggestion
static const StringToFloatConverterImpl g_string_to_float;
```